app/src/main/java/org/imozerov/streetartview/ui/explore/ExploreArtActivity.kt

```diff
@@ -60,11 +84,23 @@ class ExploreArtActivity : AppCompatActivity(), ArtObjectDetailOpener {
 
         explore_floating_action_button.setOnClickListener { openSearchView() }
         search_view.findViewById(R.id.search_close_btn).setOnClickListener { closeSearchView() }
+        viewpager.addOnPageChangeListener(object : ViewPager.SimpleOnPageChangeListener(){
+            override fun onPageSelected(position: Int) {
+                super.onPageSelected(position)
+                hideDetails()
+            }
+        })
     }
 
     override fun onStart() {
         super.onStart()
         createSearchSubscription()
+        if (ContextCompat.checkSelfPermission(this, Manifest.permission.READ_CONTACTS)
```

Comment:
As far as I understand ContextCompat will return PERMISSION_GRANTED on lower target versions. So this code won't run anyway.